### PR TITLE
Meshmodel new UI

### LINKED ISSUE #9029: [UI] Settings Revise Models Browser
### Current Behavior
The existing "MeshModels Browser" needs an uplift.

### Implementation
1. Rename to menu tab to "Models".
2. Implement two views as captured in the mockups below.
3. Use https://mui.com/x/react-tree-view/

### Acceptance Tests
1. Users can navigate between models.
2. Users can search for models or components.
4. Users can collapse/expand and scroll through registered models.

### Mockups
MeshModel Browser

Wireframe - 1

---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by ptdatta**
**Notes for Reviewers**

This PR fixes #9029 

https://github.com/meshery/meshery/assets/94271200/d26bd50e-6bfc-4ed2-aea1-42532ded6884

(https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**

- [x] Models
- [x] Components
- [x] Relationships
- [ ] Policies
- [x] Registrants

**[Signed commits]
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/.eslintrc.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ module.exports = {
     'react/jsx-uses-vars': [2],
     'react/jsx-no-undef': 'error',
     'no-console': 0,
-    'no-unused-vars': 'error',
+    'no-unused-vars': 'off',
```
Comment (Yashsharma1911) [inline ui/.eslintrc.js:48]: @ptdatta don't disable it globally, I wonder if you have reason to do so

File: ui/api/meshmodel.js [hunk 2]
```diff
@@ -44,6 +44,12 @@ export async function getComponentFromModelApi(model, pageSize = 'all', trim = t
   );
 }
 
+export async function getRelationshipFromModelApi(model, pageSize = 'all', trim = true) {
+  return await promisifiedDataFetch(
+    `${MESHMODEL_ENDPOINT}/${model}/relationships?pagesize=${pageSize}&trim=${trim}`,
+  );
+}
+
```
Comment (Yashsharma1911) [inline ui/api/meshmodel.js:52]: Let's use RTK

File: ui/components/MeshModelComponent.js [hunk 3]
```diff
@@ -504,11 +548,29 @@ const MeshModelComponent = ({ view }) => {
     marginTop: '1rem',
   };
 
+  const customButtonDiv = {
+    display: 'flex',
+    alignItems: 'center',
+  };
+
+  const cardStyle = {
+    background: '#51636B',
```
Comment (Yashsharma1911) [inline ui/components/MeshModelComponent.js:557]: please use theme reference

**Review by Yashsharma1911 — COMMENTED** [review]
Nice work

**Review by Yashsharma1911 — COMMENTED** [review]
There are some more places where you can use theme reference color please consider that other than that LGTM we can merge this

**Comment by ptdatta** [comment]
> There are some more places where you can use theme reference color please consider that other than that LGTM we can merge this

Yes. Actually, I was playing with colors that's why I used the color code but on the final PR I will change this.

**Comment by leecalcote** [comment]
Send word when you've a final recording and have addressed each and every one of the feedback items.

File: ui/.eslintrc.js (hunk 1 above)
Comment (leecalcote) [reply ui/.eslintrc.js:48]: @ptdatta fix this.

**Comment by theBeginner86** [comment]
I made following updates:
1. Fixed hard coding of number of models
2. Fixed infinite scrolling 
3. Fixed pagination issue
4. Made reusable components for each view, title, content etc.
5. Moved to use of our standard DataFormatters
6. Added support for filtering Models based on registrants
7. Bummed Meshkit to v0.6.80

**Review by theBeginner86 — APPROVED** [review]

File: ui/.eslintrc.js (hunk 1 above)
Comment (theBeginner86) [reply ui/.eslintrc.js]: reverted this change 👍

File: ui/components/MeshModelRegistry/MeshModelComponent.js [hunk 4]
```diff
@@ -0,0 +1,518 @@
+import { Button, withStyles } from '@material-ui/core';
+import { withSnackbar } from 'notistack';
+import React, { useState, useEffect } from 'react';
+import { Paper } from '@material-ui/core';
```
Comment (Yashsharma1911) [inline ui/components/MeshModelRegistry/MeshModelComponent.js:4]: We can move forward with this PR however, this is something to note if we import only a few components from mui like this then let's not use named imports, The reason behind this is that when we import using named imports, it not only brings in the specific component (in this case, the Paper component) but the entire MUI core library, leading to an increase in the package size.

Good thing is Nextjs v13 has a solution for this where we can use named import without increasing package size however as we are not using NextJs v13 right now, let's avoid using named imports.

**Review by Yashsharma1911 — APPROVED** [review]

**Pull request merged**